pkg/agent/multicluster/mc_route_controller.go

```
@@ -213,20 +242,202 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
+// any synchronization mechanism.
+func (c *MCDefaultRouteController) syncWireGuard() error {
+	gateway, err := c.getActiveGateway()
+	if err != nil {
+		return err
+	}
+
+	amIGateway := gateway != nil && gateway.Name == c.nodeConfig.Name
+	if c.wireGuardClient != nil && (!amIGateway || !c.wireGuardInitialized) {
+		if err := c.cleanUpWireGuard(); err != nil {
+			return err
+		}
+		c.wireGuardInitialized = false
+	}
+	if !amIGateway {
+		return nil
+	}
+	if !c.wireGuardInitialized {
+		var initErr error
+		defer func() {
+			if initErr != nil {
+				if err := c.cleanUpWireGuard(); err != nil {
+					klog.ErrorS(err, "Failed to clean up WireGuard")
+				}
+				c.wireGuardInitialized = false
+			}
+		}()
+		if initErr = c.initializeWireGuard(gateway); initErr != nil {
+			return initErr
+		}
+		c.wireGuardInitialized = true
+	}
+	ciImports, err := c.ciImportLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	desireCIImport := sets.NewString()
+	var updateErr error
+	for _, ciImport := range ciImports {
+		desireCIImport.Insert(ciImport.Name)
+		if err = c.updateSingleWireGuardPeer(ciImport); err != nil {
+			klog.V(2).ErrorS(err, "Failed to update WireGuard peer", "ClusterInfoImport", klog.KObj(ciImport))
+			updateErr = err
+		}
+	}
+	if updateErr != nil {
+		return updateErr
+	}
+
+	// Check cache and existing ClusterInfoImport, clean up route and WireGuard peer of the
+	// removed ClusterInfoImport.
+	installCIImports := sets.StringKeySet(c.installedCIImports)
+	for ci := range installCIImports {
+		if desireCIImport.Has(ci) {
+			continue
+		}
+		if err := c.removeWireGuardRouteAndPeer(c.installedCIImports[ci]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *MCDefaultRouteController) removeWireGuardRouteAndPeer(ciImport *mcv1alpha1.ClusterInfoImport) error {
+	remoteGatewayIP, _, err := net.ParseCIDR(ciImport.Spec.ServiceCIDR)
+	if err != nil {
+		return err
+	}
```

I think we should ignore the error if it's not a valid CIDR, it should never be installed at the very beginning.